### DISCUSSION TITLE
How to disable some of the default PMD rules that come with salesforce code analyser installation?

### DISCUSSION THREAD

**Discussion by kirankapila** [forcedotcom/sfdx-scanner#1583] [Q&A]
Hi Team,

I need to exclude some of the default PMD rules that are available with salesforce code analyser. 

I could see that most rules are spit our at runtime as .xml file from within .jar files.

Is there a better way of disabling a subset of the default PMD rules instead of meddling with .jar files? Please advise

Thanks

**Comment by stephen-carter-at-sf** [comment]
Now that `code-analyzer` has gone GA, you should check out our rule selection mechanisms. You can now add your own tags to the rules that you do wish to run and remove tags as well. So if you do not want some rules to run with the default `Recommended` tag then simple remove the `Recommended` tag from your `code-analyzer.yml` for this rule. See: https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/config-custom.html
